test/es-module/test-cjs-esm-warn.js

```diff
@@ -1,3 +1,6 @@
+// This used to test require(esm) throws ERR_REQUIRE_ESM, which is no longer the case
+// since require(esm) has been implemented. It's now repurposed to test that the
+// previous behavior can be kept when  --no-experimental-require-module is passed.
```

Comment:
```suggestion
// Previously, this tested that require(esm) throws ERR_REQUIRE_ESM, which is no longer applicable 
// since require(esm) is now supported. The test has been repurposed to ensure that the old behavior 
// is preserved when the --no-experimental-require-module flag is used.
```